### PR TITLE
[char.traits] Better cross-reference several headers

### PR DESCRIPTION
**Pull request by AlisdairM**
Add a few cross-references for headers used in the char_traits clauses.

### CODE REVIEW

File: source/strings.tex [hunk 1]
```diff
@@ -261,7 +261,7 @@
 \end{codeblock}
 
 \pnum
-The header \libheader{string}
+The header \libheaderref{string}
```
[the remark this answers is not in the document]
Comment (jensmaurer) [reply source/strings.tex:264]: I'm not enthusiastic about this particular change, but can see the value of consistency (i.e. having xrefs everywhere we mention a header that isn't in the immediate vicinity).

**Review by jensmaurer — APPROVED** [review]

File: source/strings.tex (hunk 1 above)
[the remark this answers is not in the document]
Comment (tkoeppe) [reply source/strings.tex:264]: Yes, let's revert this one. That's where editorial discretion comes in. We don't have to follow some lexical notion of consistency blindly, we can do whatever makes the document look and feel the most useful.

[the remark this answers is not in the document]
Comment (AlisdairM) [reply source/strings.tex:264]: Reverted that one change, rebased onto current `main`, and force-pushed so can merge cleanly.

[the remark this answers is not in the document]
Comment (tkoeppe) [reply source/strings.tex:264]: Thanks! Have you pushed that already? I'm not seeing it yet.

[the remark this answers is not in the document]
Comment (AlisdairM) [reply source/strings.tex]: I pushed, but had forgotten to commit my file!  Rebased onto an even more recent master and hopefully pushed the right changes this time.

**Pull request merged**